libs/x-content/src/main/java/org/elasticsearch/common/xcontent/AbstractObjectParser.java

```diff
@@ -211,6 +213,27 @@ public void declareIntArray(BiConsumer<Value, List<Integer>> consumer, ParseFiel
         declareField(consumer, (p, c) -> parseArray(p, () -> itemParser.parse(p, c)), field, type);
     }
 
+    /**
+     * Declares a set of fields that are required for parsing to succeed. Only one of the values
+     * provided per String[] must be matched.  E.g.
+     *
+     * declareRequiredFieldSet(new String[]{"foo", "bar"}); means "foo" or "bar" (or both) fields must be
+     * specified by the user.  If neither of those fields are configured, an exception will be thrown.
+     *
+     * Multiple required sets can be configured:
+     *
+     *   declareRequiredFieldSet(new String[]{"foo", "bar"});
+     *   declareRequiredFieldSet(new String[]{"bizz", "buzz"});
+     *
+     * requires that ("foo" OR "bar") AND ("bizz" OR "buzz") are configured by the user
+     */
+    public void declareRequiredFieldSet(String[] requriedSet) {
```

Comment:
This could be a varargs so the call to the method doesn't require explicitly creating an array?